### DISCUSSION TITLE
Kernel image size / improve boot speed

### DISCUSSION THREAD

**Discussion by jjalling** [OE4T/meta-tegra#587] [General]
Hi,

I have a project, currently running on a nVidia Xavier NX and I'm trying to improve to boot speed. Digging through the system I find that the kernel itself is ~300MB ! Do you see the same?
I have disabled a lot of drivers that I don't need, but it has only reduced the size 20MB or so.

What is your experience with reducing the size? If I look at my laptop kernel it's just ~10MB an I think that's big already.

BR Jonas

**Comment by madisongh** [comment]
There is some information [in the L4T documentation](https://docs.nvidia.com/jetson/l4t/index.html#page/Tegra%20Linux%20Driver%20Package%20Development%20Guide/kernel_boot_time.html) on improving kernel boot time.  The biggest win I've seen has been from reducing the kernel log level so kernel messages don't get written to the serial console.

As far as kernel size goes - make sure you're comparing apples to apples.  For one of my projects where I've customized the kernel config to eliminate unused features/drivers, the uncompressed (`vmlinux`) kernel still runs about 300MB, although I also build in almost all of the drivers I am using on that project -- including nvgpu, which is fairly large.  The compressed (`Image`) kernel is only 27MB, though.

Most desktop Linux distros make extensive use of loadable modules, which helps reduce the size of the main kernel -- even then, the compressed (`vmlinuz`) on my Ubuntu system runs 14MB.  I'm not sure that's really helpful for improving boot speed, or overall space usage, though.  Compiled x86 code tends to be more compact than ARM code, too.
